lib/workspace.js

```diff
@@ -473,22 +479,30 @@ module.exports = {
 
     const archive = archiver('zip');
     const remoteName = `workspace-${workspace_id}-${workspace_version}`;
+    const remoteDir = path.join(config.workspaceHomeDirRoot, remoteName, 'current');
+    const gradedFiles = await fg(workspace_graded_files, { cwd: remoteDir, stats: true });
+
+    if (gradedFiles.length > config.workspaceMaxGradedFilesCount) {
+      throw { message: 'Too many files submitted in workspace.', isFormatError: true };
```

Comment:
Throwing a non-error object is inadvisable. If you want to be able to identify a specific kind of error, I'd recommend subclassing `Error` (e.g. `SubmissionFormatError`) and then checking if an error is an instance of that.

Reply:
Fixed in https://github.com/PrairieLearn/PrairieLearn/commit/8425f6166fb906eeb0287fb35eff2266737c1793.